### PR TITLE
installation: CNF Installation (8) Transition to new installation

### PR DESCRIPTION
**Pull request by kosstennbl**
## Description
Remove old installation method, use the new one instead.
Change specs accordingly.
Remove unnecessary config parameters.
Fully enable multiple deployments installation.

## Issues:
Refs: #2169

### CODE REVIEW

File: src/tasks/utils/cnf_manager.cr [hunk 1]
```diff
@@ -203,407 +182,8 @@ module CNFManager
...
-    when CNFInstall::InstallMethod::ManifestDirectory
-      manifest_directory = config.deployments.get_deployment_param(:manifest_directory)
-      installed_manifest_directory = File.join(destination_cnf_dir, manifest_directory)
-      Log.for("cnf_cleanup:installed_manifest_directory").info { installed_manifest_directory }
-      result = KubectlClient::Delete.file("#{installed_manifest_directory}", wait: true)
-      FileUtils.rm_rf(destination_cnf_dir)
-      stdout_success "Successfully cleaned up #{manifest_directory} directory"
-      return true
-    end
+    # (kosstennbl) TODO: Redesign this method using new installation.
```
Comment (svteb) [inline src/tasks/utils/cnf_manager.cr:186]: Let's not forget to write a ticket for this.

**Review by svteb — COMMENTED** [review]
As we've discussed in a meeting, I think that rewriting all mentions of cnf_setup/cnf_cleanup internally to install_cnf/uninstall_cnf might be hard to understand for new contributors. Seeing the state of the code I might be more inclined to switch the cnf_setup/cnf_cleanup to cnf_install/cnf_uninstall on the terminal more now, but I am still not sure (@martin-mat we could use your input on this matter). Additionally I do not think that it is necessary to name the methods install_cnf/uninstall_cnf when they are part of CNFInstall module.

File: src/tasks/utils/cnf_manager.cr (hunk 1 above)
Comment (kosstennbl) [reply src/tasks/utils/cnf_manager.cr:186]: Done

**Review by svteb — APPROVED** [review]
lgtm

File: spec/5g/core_spec.cr [hunk 2]
```diff
@@ -18,7 +18,7 @@ describe "Core" do
       result = ShellCmd.run_testsuite("smf_upf_heartbeat verbose")
       (/(PASSED).*(Chaos service degradation is less than 50%)/ =~ result[:output]).should_not be_nil
     ensure
-      result = ShellCmd.run_testsuite("cnf_cleanup cnf-config=sample-cnfs/sample_open5gs/cnf-testsuite.yml") 
+      result = ShellCmd.cnf_cleanup()
```
Comment (collivier) [inline spec/5g/core_spec.cr:21]:
```suggestion
      result = ShellCmd.cnf_cleanup()
```

File: spec/5g/core_spec.cr [hunk 3]
```diff
@@ -29,7 +29,7 @@ describe "Core" do
       result = ShellCmd.run_testsuite("smf_upf_heartbeat verbose baseline_count=300")
       (/(FAILED).*(Chaos service degradation is more than 50%)/ =~ result[:output]).should_not be_nil
     ensure
-      result = ShellCmd.run_testsuite("cnf_cleanup cnf-config=sample-cnfs/sample_open5gs/cnf-testsuite.yml") 
+      result = ShellCmd.cnf_cleanup()
```
Comment (collivier) [inline spec/5g/core_spec.cr:32]:
```suggestion
      result = ShellCmd.cnf_cleanup()
```

File: spec/5g/core_spec.cr [hunk 4]
```diff
@@ -40,7 +40,7 @@ describe "Core" do
       result = ShellCmd.run_testsuite("suci_enabled verbose")
       (/(PASSED).*(Core uses SUCI 5g authentication)/ =~ result[:output]).should_not be_nil
     ensure
-      result = ShellCmd.run_testsuite("cnf_cleanup cnf-config=sample-cnfs/sample_open5gs/cnf-testsuite.yml") 
+      result = ShellCmd.cnf_cleanup()
```
Comment (collivier) [inline spec/5g/core_spec.cr:43]:
```suggestion
      result = ShellCmd.cnf_cleanup()
```

File: spec/5g/core_spec.cr [hunk 5]
```diff
@@ -51,7 +51,7 @@ describe "Core" do
       result = ShellCmd.run_testsuite("suci_enabled verbose")
       (/(FAILED).*(Core does not use SUCI 5g authentication)/ =~ result[:output]).should_not be_nil
     ensure
-      result = ShellCmd.run_testsuite("cnf_cleanup cnf-config=sample-cnfs/sample_open5gs_no_auth/cnf-testsuite.yml") 
+      result = ShellCmd.cnf_cleanup()
```
Comment (collivier) [inline spec/5g/core_spec.cr:54]:
```suggestion
      result = ShellCmd.cnf_cleanup()
```

File: spec/utils/cnf_manager_spec.cr [hunk 6]
```diff
@@ -319,30 +174,33 @@ describe "SampleUtils" do
...
   end
 
   it "'CNFManager.workload_resource_test' should accept an args and cnf-config argument, populate a deployment, container, and intialized argument, and then apply a test to a cnf", tags: ["cnf-config"]  do
-    args = Sam::Args.new(["cnf-config=./sample-cnfs/sample-generic-cnf/cnf-testsuite.yml"])
-    cli_hash = CNFManager.sample_setup_cli_args(args, false)
-    CNFManager.sample_setup(cli_hash) if cli_hash["config_file"]
-    config = CNFInstall::Config.parse_cnf_config_from_file("./sample-cnfs/sample-generic-cnf/cnf-testsuite.yml")    
-    task_response = CNFManager.workload_resource_test(args, config) do |resource, container, initialized|
-      test_passed = true
-				begin
```
Comment (collivier) [inline spec/utils/cnf_manager_spec.cr:334]: please format

File: spec/utils/utils_spec.cr [hunk 7]
```diff
@@ -73,49 +72,52 @@ describe "Utils" do
...
-        # Only check the containers that are in the deployed helm chart or manifest
-        (privileged_list & (resource_containers_list - white_list_container_names)).each do |x|
-          violation_list << x
+  # TODO (kosstennbl) Rework this spec, shouldn't contain a real test.
+  it "'single_task_runner' should accept a cnf-config argument and apply a test to that cnf", tags: ["task_runner"] do
+    begin
+      args = Sam::Args.new()
+      config_path = "./sample-cnfs/sample-generic-cnf/cnf-testsuite.yml"
+      ShellCmd.cnf_setup("cnf-config=#{config_path}")
+
```
Comment (collivier) [inline spec/utils/utils_spec.cr:81]:
```suggestion

```

File: spec/utils/utils_spec.cr [hunk 8]
```diff
@@ -73,49 +72,52 @@ describe "Utils" do
...
-        (privileged_list & (resource_containers_list - white_list_container_names)).each do |x|
-          violation_list << x
+  # TODO (kosstennbl) Rework this spec, shouldn't contain a real test.
+  it "'single_task_runner' should accept a cnf-config argument and apply a test to that cnf", tags: ["task_runner"] do
+    begin
+      args = Sam::Args.new()
+      config_path = "./sample-cnfs/sample-generic-cnf/cnf-testsuite.yml"
+      ShellCmd.cnf_setup("cnf-config=#{config_path}")
+      
+      task_response = CNFManager::Task.single_task_runner(args) do |args, config|
```
Comment (collivier) [inline spec/utils/utils_spec.cr:82]:
```suggestion
      task_response = CNFManager::Task.single_task_runner(args) do |args, config|
```

**Review by collivier — CHANGES_REQUESTED** [review]
blank chars + format

**Comment by kosstennbl** [comment]
Whitespaces and formatting were fixed in mentioned cases, i think that there is still a lot of problems of this type in testsuite, maybe different ticket should be created to fix this.

**Review by collivier — APPROVED** [review]

**Pull request merged**